agent/app/service/container.go

```diff
@@ -794,87 +792,87 @@ func (u *ContainerService) ContainerLogClean(req dto.OperationWithName) error {
 	}
-	commandName := "docker"
-	commandArg := []string{"logs", container}
-	if containerType == "compose" {
-		commandArg = []string{"compose", "-f", container, "logs"}
+	cmdArgs = append(cmdArgs, "logs")
+	if params.Follow {
+		cmdArgs = append(cmdArgs, "-f")
 	}
-	if tail != "0" {
-		commandArg = append(commandArg, "--tail")
-		commandArg = append(commandArg, tail)
+	if params.Tail != "all" {
+		cmdArgs = append(cmdArgs, "--tail", params.Tail)
 	}
-	if since != "all" {
-		commandArg = append(commandArg, "--since")
-		commandArg = append(commandArg, since)
+	if params.Since != "all" {
+		cmdArgs = append(cmdArgs, "--since", params.Since)
 	}
-	if follow {
-		commandArg = append(commandArg, "-f")
-	}
-	if !follow {
-		cmd := exec.Command(commandName, commandArg...)
-		cmd.Stderr = cmd.Stdout
-		stdout, _ := cmd.CombinedOutput()
-		if !utf8.Valid(stdout) {
-			return errors.New("invalid utf8")
-		}
-		if err := wsConn.WriteMessage(websocket.TextMessage, stdout); err != nil {
-			global.LOG.Errorf("send message with log to ws failed, err: %v", err)
-		}
-		return nil
+	if params.Container != "" {
+		cmdArgs = append(cmdArgs, params.Container)
 	}
+	cmd := exec.Command("docker", cmdArgs...)
 
-	cmd := exec.Command(commandName, commandArg...)
 	stdout, err := cmd.StdoutPipe()
 	if err != nil {
-		_ = cmd.Process.Signal(syscall.SIGTERM)
-		return err
+		errorChan <- fmt.Errorf("failed to get stdout pipe: %v", err)
+		return
 	}
-	cmd.Stderr = cmd.Stdout
 	if err := cmd.Start(); err != nil {
-		_ = cmd.Process.Signal(syscall.SIGTERM)
-		return err
+		errorChan <- fmt.Errorf("failed to start command: %v", err)
+		return
 	}
-	exitCh := make(chan struct{})
-	go func() {
-		_, wsData, _ := wsConn.ReadMessage()
-		if string(wsData) == "close conn" {
-			_ = cmd.Process.Signal(syscall.SIGTERM)
-			exitCh <- struct{}{}
-		}
-	}()
 
-	go func() {
-		buffer := make([]byte, 1024)
-		for {
-			select {
-			case <-exitCh:
-				return
-			default:
-				n, err := stdout.Read(buffer)
-				if err != nil {
-					if err == io.EOF {
-						return
-					}
-					global.LOG.Errorf("read bytes from log failed, err: %v", err)
-					return
-				}
-				if !utf8.Valid(buffer[:n]) {
-					continue
-				}
-				if err = wsConn.WriteMessage(websocket.TextMessage, buffer[:n]); err != nil {
-					global.LOG.Errorf("send message with log to ws failed, err: %v", err)
-					return
-				}
-			}
+	scanner := bufio.NewScanner(stdout)
+	lineNumber := 0
+
+	for scanner.Scan() {
+		lineNumber++
+		message := scanner.Text()
+		select {
+		case messageChan <- message:
+		case <-time.After(time.Second):
+			errorChan <- fmt.Errorf("message channel blocked")
+			return
 		}
-	}()
-	_ = cmd.Wait()
-	return nil
+	}
+
+	if err := scanner.Err(); err != nil {
+		errorChan <- fmt.Errorf("scanner error: %v", err)
+		return
+	}
+	cmd.Wait()
 }
 
 func (u *ContainerService) DownloadContainerLogs(containerType, container, since, tail string, c *gin.Context) error {
```

Comment:
The code you provided is from an implementation of Docker Container Service. There doesn't seem to be any issues mentioned in the comments about irregularities, potential issues or opportunities for optimization. The usage makes sense based on docker's commands such as `container logs` which are part of standard operations performed by docker containers.

However, there could still be some room to improve readability and simplicity through better naming conventions, more descriptive docstrings, etc., especially when dealing with complex use cases.

For example: 

Change `-command := exec.Command(commandName, commandArg...) // this can lead to unexpected behavior`
to `cmd := &exec.Cmd{ ... }`
In general it would be beneficial if someone looking at the code has a thorough understanding of how the function works first before diving into detailed optimizations/coding standards adjustments.

Also consider refactoring functions like downloading logs to handle other types of data like images, networks etc., depending upon application requirements:

```
func (u *ContainerService) DownloadImageLogs(image string, version string, versionID int64) error
    func (u *ContainerService) DownloadNetworkLogs(network string, since string, tail string, containerType bool) error

...
```